components/LeadFormDialog.tsx

```diff
@@ -0,0 +1,157 @@
+import * as React from "react";
+import Button from "@mui/material/Button";
+import TextField from "@mui/material/TextField";
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogTitle from "@mui/material/DialogTitle";
+import MenuItem from "@mui/material/MenuItem";
+import { Lead, LeadStatus } from "@prisma/client";
+
+interface ErrorDisplayProps {
+  field: string;
+  errors?: { key: string; message: string }[];
+}
+
+function ErrorDisplay({ field, errors }: ErrorDisplayProps) {
+  return (
+    <>
+      {errors &&
+        errors.map((error) => {
+          if (error.key === field) {
+            return (
+              <div key={error.key} className="text-red-500">
+                {error.message}
+              </div>
+            );
+          }
+          return null;
+        })}
+    </>
+  );
+}
+
+interface LeadFormDialogProps {
+  leadData?: Lead;
+  onSubmit: (lead: Lead) => Promise<void>;
+  onClose: () => void;
+  errors?: { key: string; message: string }[];
+}
+
+export default function LeadFormDialog({
+  leadData,
+  onSubmit,
+  onClose,
+  errors,
+}: LeadFormDialogProps) {
+  const [isLoading, setIsLoading] = React.useState(false);
+
+  const [lead, setLead] = React.useState<Lead>(
+    leadData ||
+      ({
+        name: "",
+        email: "",
+        status: "PROSPECT" as LeadStatus,
+        estimatedSaleAmount: 0,
+      } as Lead)
+  );
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setLead({
+      ...lead,
+      [name]: value,
+    });
+  };
+
+  return (
+    <div>
+      <Dialog
+        open
+        onClose={(event, reason) => {
+          if (reason && reason === "backdropClick") return;
+          onClose();
+        }}
+      >
+        <DialogTitle className="text-lg font-medium">
+          {leadData ? "Edit Lead" : "Create Lead"}
+        </DialogTitle>
+        <DialogContent className="px-4 py-2 h-[400px] w-[600px]">
+          <TextField
```

Comment:
These fields should have basic client side validation that mirrors the server side validation for most things unless it has to be determined server side so that users dont have to wait for an http request to complete to see their error.

This wouldn't be hard since we essentially have a full TS mono-repo and can share validation code between the frontend and backend to unify the validation. But with the time given I just drove the errors completely from the backend and mapped the error responses to the appropriate fields